### PR TITLE
chore: All the cleaning 🧹

### LINKED ISSUE #147: Rename threshold percentile to positive rate
Todo:

* [ ] `sens_over_time` heatmap: Rename "threshold percentile" to "positive rate" and calculate as 100-threshold_percentile

* [ ] `performance_by_threshold` table: do the same

### LINKED ISSUE #164: Suggestion: Add docstring formatter
The last few reviews (e.g., #148) had errors mainly related to docstrings.

Therefore I suggest adding [pycodestyle](https://github.com/PyCQA/pycodestyle) and [darglint](https://github.com/terrencepreilly/darglint) to our pre-commit hooks.

pycodestyle check for style conventions in the docstring (initial tests seem promising, although potentially too strict). darglint checks whether the description matches the implementation (e.g., type hints consistency).

This should reduce 1) time spent on code review, 2) reduce cycles in code review (see #148) 3) Remove inconsistencies between code and documentation.

Couldn't I find an autoformatter, sadly (darglint has it on its roadmap).

Suggestion for darglint settings:
in file `.darglint`:
```
[darglint]
ignore=DAR401
strictness=short
```

Potential frustrations with darglint:
Does not support ", optional" type hints:
```python
# valid
test_size (float): test_size is the test size...
# not valid
test_size (float, optional): test_size is the test size...
# but is valid (if type hint match)
test_size (Optional[float]): test_size is the test size...

```
you can choose to ignore: DAR103

Adding to pre-commit (flake8-docstrings == pycodestyle):
```

  - repo: https://github.com/pycqa/pydocstyle
    rev: 6.1.1
    hooks:
      - id: pydocstyle
        files: ^src/
        args:
        - --ignore=D203,D204,D213,D215,D400,D401,D404,D406,D407,D408,D409,D413,D104
      # ignore corresponds to google docstring conventions
      # with the additions of: 
      # D104: require docstring in __init__.py

  - repo: https://github.com/terrencepreilly/darglint
    rev: v1.8.1
    hooks:
      - id: darglint
```

### PR DESCRIPTION
**Pull request by MartinBernstorff**
Fixes #164 . See the issue for discussion of choices.

When reviewing, don't review the `.py` files. They all follow from the linting.

### CODE REVIEW

File: .darglint [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[darglint]
+ignore=DAR003, DAR401
+strictness=short
+indentation=4
+
+# Removing the object id from the error message allows the path to be Cmd+Click-able in VSCode. 
+# Makes it much easier to navigate to the line.
+# (E.g. src/psycopt2d/data/generate_synthetic_dataset_for_eval.py:overwrite_prop_with_null:54 becomes 
+# src/psycopt2d/data/generate_synthetic_dataset_for_eval.py:54). 
+message_template= - {path}:{line}: {msg} ({msg_id} in {obj})
```
Comment (KennethEnevoldsen) [inline .darglint:10]: Love this thanks.

File: pyproject.toml [hunk 2]
```diff
@@ -33,6 +33,8 @@ ipython = ">=8.4.0,<8.5.0"
 ipykernel = ">=6.15.1,<6.16.0"
 flake8 = ">=4.0.1,<4.1.0"
 pytest-xdist = "^2.5.0"
+pydocstyle = "^6.1.1"
```
Comment (HLasse) [inline pyproject.toml:36]:
```suggestion
```

**Review by KennethEnevoldsen — APPROVED** [review]
Hmm, I am unsure if I would prefer the pre-commit instead. It simplifies the CI and ensures that if you push it, you are pretty good to go. This might still lead to more development cycles where you push, see that it fails, and push again (instead of fixing it locally before pushing).

That being said, it is annoyingly slow. So let us just go with this. We can always change it if needed.

File: .darglint [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[darglint]
+ignore=DAR003, DAR401
```
Comment (KennethEnevoldsen) [inline .darglint:2]: Should DAR003 be ignored?

Comment (MartinBernstorff) [reply .darglint:2]: It doesn't seem consistent – I tried setting indent manually and it still didn't work well, so decided to ignore instead.

**Comment by MartinBernstorff** [comment]
> Hmm, I am unsure if I would prefer the pre-commit instead. It simplifies the CI and ensures that if you push it, you are pretty good to go. This might still lead to more development cycles where you push, see that it fails, and push again (instead of fixing it locally before pushing).
> 
> That being said, it is annoyingly slow. So let us just go with this. We can always change it if needed.

I strongly agree. If it were faster, pre-commit would be the way to go. 

I still think this is better than nothing, though. Most of the stuff darglint caught were meaningful changes to documentation/interface. If it gets too annoying, we'll have to remove it, though.

**Pull request merged**